### DISCUSSION TITLE
Curso Modelando Domínios Ricos - Usando o Domínio em uma API

### DISCUSSION THREAD

**Discussion by KIQxl** [balta-io/forum#551] [Tira Dúvidas] [answered]
Boa tarde! 
Recentemente terminei o curso de modelagem de domínios ricos e gostaria de saber como faço para utilizar a dll que foi gerada através do projeto domain em uma api por exemplo. Como faria para unir esses projetos e para que eles se comunicassem ou consumissem um ao outro?

**Comment by henriqueweege** [comment, marked as answer]
Oi, @KIQxl! Tudo bem?
Então, voce pode seguir [este artigo](https://balta.io/blog/github-packages-github-actions-distribuindo-contextos-delimitados) do Balta que explica como publicar pacotes no Nuget. A partir disso, funciona como qualquer outro pacote. Essa é a melhor solução.

Outra possibilidade, mais "dirty" mas que pode servir para o teu propósito, é copiar as dlls e colocar em alguma pasta específica e conhecida,  e a partir desta pasta adicionar o pacote. Você tem como fazer isso direto pelo VS Studio.

Basta clicar com o botão direito do mouse nas dependências do projeto e ai clicar em "Adicionar Referência de Projeto"
![image](https://github.com/balta-io/forum/assets/96916801/fd9d9bd3-bd1d-4226-b832-fe84357a0170)

Então em "Procurar":
![image](https://github.com/balta-io/forum/assets/96916801/d5174922-0243-40a9-8a8f-6ccbd18a477f)

E ai você acha e adiciona a dll que você quer:
![image](https://github.com/balta-io/forum/assets/96916801/64a8afd4-e25a-48de-83a2-dd9a3dbc8440)
![image](https://github.com/balta-io/forum/assets/96916801/dc4c734c-6805-4e8b-ae0d-abe39b062101)

Agora você tem disponível as classes que estão na dll:
![image](https://github.com/balta-io/forum/assets/96916801/24a645e9-54ef-48c2-bc85-653e18acb7ec)



Essa solução não é das melhores principalmente se pensa em ter outras versões da dll em questão. Para um teste rápido ou algo não muito sério é uma solução válida. Eu não usaria isso em produção. Trabalhei em um equipe que funcionava dessa maneira, e acredite em mim, colocar dll no servidor na mão as 22:00 de um sexta-feira depois de uma semana pesada é de tirar o sono hahaha 

Valeeeeu!

**Reply by KIQxl** [reply]
Show de bola!
Muito obrigado pela explicação.
